google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineHelper.java

```diff
@@ -16,71 +16,103 @@
 
 package com.google.cloud.tools.intellij.appengine.cloud;
 
+import com.google.cloud.tools.app.impl.cloudsdk.internal.process.ProcessExitListener;
+import com.google.cloud.tools.app.impl.cloudsdk.internal.process.ProcessOutputLineListener;
+import com.google.cloud.tools.app.impl.cloudsdk.internal.process.ProcessStartListener;
+import com.google.cloud.tools.app.impl.cloudsdk.internal.sdk.CloudSdk;
+import com.google.cloud.tools.intellij.appengine.cloud.CloudSdkAppEngineHelper.Environment;
+
 import com.intellij.openapi.project.Project;
-import com.intellij.remoteServer.runtime.deployment.DeploymentRuntime.UndeploymentTaskCallback;
+import com.intellij.openapi.vcs.impl.CancellableRunnable;
 import com.intellij.remoteServer.runtime.deployment.ServerRuntimeInstance.DeploymentOperationCallback;
 import com.intellij.remoteServer.runtime.log.LoggingHandler;
 
 import java.io.File;
+import java.io.IOException;
 
 /**
  * Provides basic Gcloud based App Engine functionality for our Cloud Tools plugin.
  */
 public interface AppEngineHelper {
 
   /**
+   * The project within the context of this helper.
+   */
+  Project getProject();
+
+  /**
    * The path to the gcloud command on the local file system.
    */
   File getGcloudCommandPath();
 
   /**
+   * The App Engine environment.
+   */
+  Environment getEnvironment();
```

Comment:
as mentioned earlier I'm wondering whether this is the right scope for 'Environment'.